### ISSUE TITLE
Logistic Regression Algorithm

### ISSUE THREAD

**Issue by justwagle** [STEllAR-GROUP/phylanx#23]
As a first step towards implementing Logistic Regression in Phylanx, linked below is a simple implementation of the logistic regression algorithm in Python. 

https://gist.github.com/justwagle/0a75572e7b34252916dc9a5c894a4ef0

Primitives required for the above algorithm are as follows :   

- [x] N Dimensional Array
- [x] Transpose Operation
- [x] Dot Operation
- [x] + , - , * , /  Operations
- [x] assignment (`store()`)
- [x] Exponential Calculation 
- [x] zero-initialized arrays (or initialized with any value)

**Comment by hkaiser** [comment]
> Primitives required for the above algorithm are as follows :

@justwagle two questions:

- Is it ok to stick to 0-2 dimension for this for now? I.e. does it make sense to have this algorithm implemented for 0-2 dimensional data?
- Should we rely on numpy's internal implementation and use this directly? Would Eigen3 be a viable option?

**Comment by justwagle** [comment]
I think it should be OK to have this implemented for 0-2 dimensional data for now. As far as using numpy's internal implementation or Eigen3, I need to put more thought into that.

**Comment by ct-clmsn** [comment]
@justwagle nice! as an aside, here are two similarly structured solutions as a point of comparison against scikit learn: [one](https://stackoverflow.com/questions/17784587/gradient-descent-using-python-and-numpy), [two](https://github.com/mattnedrich/GradientDescentExample/blob/master/gradient_descent_example.py)

**Comment by hkaiser** [comment]
FWIW, the LR-algorithm as sketched out by @justwagle above could be represented using the currently available (and planned) primitives as:

```
block(
	store(iterations, 750),
	store(alpha, 1e-5),
	store(x, file_read("breast_cancer_data")),
	store(y, file_read("breast_cancer_target_data")),
	store(w, constant(0, x)),
	store(transx, transpose(x)),
	while(
		step < iterations,
		block(
			store(pred, 1.0 / (1.0 + exp(dot(x, w)))),
			store(error, pred - y),
			store(gradient, dot(transx, error)),
			store(w, w - (alpha * gradient)),
			store(step, step + 1)
		)
	),
	file_write("weights", w)
)
```

**Comment by hkaiser** [comment]
This was fixed by merging #62
